src/controller/block_editor_controller.js

```diff
@@ -348,26 +366,24 @@ class BlockEditorController {
     // Backup Blockly.Blocks object so that main workspace and preview don't
     // collide if user creates a 'factory_base' block, for instance.
     const backupBlocks = Blockly.Blocks;
+    const tempBlockType =  '____temporary_block_type_name_for_preview';
+    const tempDefinition = new BlockDefinition(tempBlockType, format, code);
```

Comment:
I know you said this is what you were doing in the CL description, but I don't quite understand. Why is using the existing blockDefinition a problem to generate the preview?

Reply:
The manual code may be an incomplete edit, and might not compile. Overwriting the model with this interim edit means we've lost the version that can generate a preview block and export to a functional real block.

Reply:
Got it. So this is the same as the other dev tools now. LGTM.

Though, I think in the long run it would be better to give the user the error message if it doesn't parse and not allow them to exit manual mode.

Reply:
I added an issue number to that TODO.